bindings/python/FloatingBaseEstimators/src/FloatingBaseEstimators.cpp

```diff
@@ -26,7 +26,48 @@ void CreateKinDynComputations(pybind11::module& module)
     namespace py = ::pybind11;
     using namespace iDynTree;
     py::class_<KinDynComputations, std::shared_ptr<KinDynComputations>>(module,
-                                                                        "KinDynComputations");
+                                                                        "KinDynComputations")
+         .def("get_nr_of_dofs", [](const KinDynComputations& impl) {
+                return impl.getNrOfDegreesOfFreedom();
+            })
+         .def("get_joint_pos", [](const KinDynComputations& impl) {
+                Eigen::VectorXd s(impl.getNrOfDegreesOfFreedom());
+                if (!impl.getJointPos(s))
+                {
+                    throw py::value_error("Failed to get the joint position.");
+                }
+                return s;
+            })
+        .def("set_joint_pos", [](KinDynComputations& impl, Eigen::Ref<const Eigen::VectorXd> s) {
+                return impl.setJointPos(iDynTree::make_span(s.data(), s.size()));
+            },
+            py::arg("s"))
+         .def("get_robot_state", [](KinDynComputations& impl) {
+                Eigen::Matrix<double, 4, 4> world_T_base;
+                Eigen::VectorXd s(impl.getNrOfDegreesOfFreedom());
+                Eigen::VectorXd base_velocity(6);
+                Eigen::VectorXd s_dot(impl.getNrOfDegreesOfFreedom());
+                Eigen::VectorXd world_gravity(3);
+                if (!impl.getRobotState(world_T_base,s,base_velocity,s_dot,world_gravity))
+                {
+                    throw py::value_error("Failed to get the robot state.");
+                }
+                py::dict robot_state;
+                robot_state["world_T_base"] = world_T_base;
+                robot_state["s"] = s;
+                robot_state["base_velocity"] = base_velocity;
+                robot_state["s_dot"] = s_dot;
+                robot_state["world_gravity"] = world_gravity;
```

Comment:
cc @GiulioRomualdi

Reply:
Since pybind11 does not support `NamedTuple` (https://github.com/pybind/pybind11/issues/1244) not `dataclasses.dataclass` (https://github.com/pybind/pybind11/issues/2065), in order to have a real return type a possibility is to define a struct in the C++ bindings. Something like this:

```cpp
struct KinDynKinematicsRobotState
{
    Eigen::VectorXd jointPositions;
    Eigen::VectorXd jointVelocities;

    Eigen::Vector3d basePosition;
    Eigen::Vector6d baseVelocity;
    Eigen::Matrix<double, 4, 4> baseTransform; // Or: Eigen::Vector4d baseQuaternion, or a manif type
    
    Eigen::Vector3d worldGravity;
};
```

Possibly this struct could be constructed with default parameters, and then each field overridden:

```py
robot_state = blf.estimators.KinDynKinematicsRobotState()
robot_state.base_position = [0, 0, 0.52]
robot_state.joint_positions = np.zeros(dofs)
robot_state.joint_velocities = np.zeros_like(robot_state.joint_positions)
```

Or, another possibility, passing a `dofs` constructor to initialize the sizes of the joint buffers:

```py
robot_state = blf.estimators.KinDynKinematicsRobotState(dofs=23)  # default dofs=0
```

Reply:
If we agree on one of the use of the struct, I will modify the PR accordingly.

Reply:
Let's wait @GiulioRomualdi opinion on this, my current preference would go with a struct that could be initialized with the dofs. And, with gravity initialized with the default value from the math component.

Reply:
I guess you can proceed with something similar. Then, if we want to change something, we'll let you know :)

```cpp
    struct KinDynKinematicsRobotState
    {
        KinDynKinematicsRobotState(const size_t dofs = 0,
                                   const Eigen::Vector3d gravity
                                   = {0,
                                      0,
                                      -BipedalLocomotion::Math::StandardAccelerationOfGravitation})
            : worldGravity(gravity)
        {
            this->jointPositions = Eigen::VectorXd::Zero(dofs);
            this->jointVelocities = Eigen::VectorXd::Zero(dofs);
        }

        Eigen::VectorXd jointPositions;
        Eigen::VectorXd jointVelocities;

        Eigen::Matrix<double, 6, 1> baseVelocity = Eigen::Matrix<double, 6, 1>::Zero();
        Eigen::Transform<double, 3, Eigen::TransformTraits::Affine> baseTransform
            = Eigen::Transform<double, 3, Eigen::TransformTraits::Affine>::Identity();

        Eigen::Vector3d worldGravity = {0, 0, -Math::StandardAccelerationOfGravitation};
    };
```

Reply:
Dictionary replaced with the struct in 38eb1ade301cce5ff1a5eb4dd79b72da159d3475.